code/modules/surgery/limb_reattach.dm

```diff
@@ -51,79 +98,87 @@
+	target.apply_damage(10, BRUTE, null, sharp=1)
+	return 0
 
 /datum/surgery_step/limb/mechanize
+	name = "attach robotic limb"
 	allowed_tools = list(/obj/item/robot_parts = 100)
 
-	min_duration = 80
 	max_duration = 100
 
-	can_use(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool)
-		if(..())
-			var/obj/item/robot_parts/p = tool
-			if (p.part)
-				if (!(target_zone in p.part))
-					return 0
-			return isnull(target.get_organ(target_zone))
-
-	begin_step(mob/user, mob/living/carbon/human/target, target_zone, obj/item/tool)
+/datum/surgery_step/limb/mechanize/can_use(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool)
+	if(..())
+		var/obj/item/robot_parts/p = tool
+		if (p.part)
+			if (!(target_zone in p.part))
+				return 0
+		return isnull(target.get_organ(target_zone))
+
+/datum/surgery_step/limb/mechanize/begin_step(mob/user, mob/living/carbon/human/target, target_zone, obj/item/tool)
 		user.visible_message("[user] starts attaching \the [tool] to [target].", \
 		"You start attaching \the [tool] to [target].")
 
-	end_step(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool)
-		var/obj/item/robot_parts/L = tool
-		user.visible_message("<span class='notice'>[user] has attached \the [tool] to [target].</span>",	\
-		"<span class='notice'>You have attached \the [tool] to [target].</span>")
-
-		if(L.part)
-			for(var/part_name in L.part)
-				if(!isnull(target.get_organ(part_name)))
-					continue
-				var/list/organ_data = target.species.has_limbs["[part_name]"]
-				if(!organ_data)
-					continue
-				var/new_limb_type = organ_data["path"]
-				var/obj/item/organ/external/new_limb = new new_limb_type(target)
-				new_limb.robotize(L.model_info)
-				if(L.sabotaged)
-					new_limb.sabotaged = 1
-
-		target.update_body()
-		target.updatehealth()
-		target.UpdateDamageIcon()
-
-		qdel(tool)
-
-	fail_step(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool)
-		user.visible_message("<span class='alert'>[user]'s hand slips, damaging [target]'s flesh!</span>", \
-		"<span class='alert'>Your hand slips, damaging [target]'s flesh!</span>")
-		target.apply_damage(10, BRUTE, null, sharp=1)
+/datum/surgery_step/limb/mechanize/end_step(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool)
+	var/obj/item/robot_parts/L = tool
+	user.visible_message("<span class='notice'>[user] has attached \the [tool] to [target].</span>",	\
+	"<span class='notice'>You have attached \the [tool] to [target].</span>")
+
+	if(L.part)
+		for(var/part_name in L.part)
+			if(!isnull(target.get_organ(part_name)))
+				continue
+			var/list/organ_data = target.species.has_limbs["[part_name]"]
+			if(!organ_data)
+				continue
+			var/new_limb_type = organ_data["path"]
+			var/obj/item/organ/external/new_limb = new new_limb_type(target)
+			new_limb.robotize(L.model_info)
+			new_limb.replaced(target)
+			new_limb.status &= ~ORGAN_DESTROYED
```

Comment:
My bad (note to self, don't start reading code before you're on your second cup of caffiene)

I'll have another look, as whilst @Crazylemon64 is right that qdeling the stump would probably work, there isn't a qdel in the original so something else has changed to cause the problem and it could cause you other problems further down the line